src/pages/users/user-discount-request.component.js

```diff
@@ -0,0 +1,53 @@
+import Template from 'components/template/template.component';
+import useDiscountRequest from 'hooks/use-discount-request';
+import useCompanies from 'hooks/use-companies';
+import DiscountRequestTable from './components/discont-request-table.component';
+import useCompaniesSelectionMenuItem from 'hooks/use-companies-selection-menu-item';
+import { TextField, MenuItem } from '@material-ui/core';
+import { CircularProgress } from '@material-ui/core';
+
+const DiscountRequests = () => {
+  const { companies, isLoading: isLoadingCompanies } = useCompanies();
+
+  const { selectedCompany, handleChangeCompany } =
+    useCompaniesSelectionMenuItem(companies);
+
+  const { invoicesList, isLoading: isLoadingDiscountRequest } =
+    useDiscountRequest(selectedCompany);
+
+  const render = () => {
+    if (isLoadingCompanies)
+      return (
+        <Template>
+          <CircularProgress testid='company-edit-spinner' size={25} />
+        </Template>
+      );
+
+    return (
+      <Template title='Pedidos de desconto'>
+        <TextField
+          label='Empresa'
+          select
+          onChange={handleChangeCompany}
+          value={selectedCompany}
+        >
+          {companies.map((company) => (
+            <MenuItem key={company.id} value={company.id}>
+              {company.name}
+            </MenuItem>
+          ))}
+        </TextField>
+        {isLoadingDiscountRequest && (
+          <CircularProgress testid='company-edit-spinner' size={25} />
+        )}
+        {!isLoadingDiscountRequest && (
+          <DiscountRequestTable invoicesList={invoicesList} />
+        )}
```

Comment:
NIT: I feel like this could be an if-else in oder to simplify the code a little bit:
```suggestion
        {isLoadingDiscountRequest ? (
          <CircularProgress testid='company-edit-spinner' size={25} />
        ) : (
          <DiscountRequestTable invoicesList={invoicesList} />
        )}
```

Reply:
I usually like to use this pattern you have used when the variable has more than two possible values. For example, if a `status` variable could be "active", "pending", and "archived", I would do:
```jsx
<>
  {status === 'active' && <p>Active</p>}
  {status === 'pending' && <p>Pending</p>}
  {status === 'archived' && <p>Archived</p>}
</>
```

If the variable has only two possible values, I like to do:
```jsx
<>
  {isLoading ? (
    <p>Loading</p>
  ) : (
    <p>Not Loading</p>
  )}
</>
```